src/Uno.UI/UI/Xaml/FrameworkElement.Layout.netstd.cs

```diff
@@ -301,25 +306,21 @@ private void ArrangeNative(Point offset, Rect clippedFrame = default)
 
 			Rect? getClip()
 			{
-				// Clip transform not supported yet on Wasm
+				// Clip transform not supported yet on Wasm/Skia
 
-				if (RequiresClipping) // if control should be clipped by layout constrains
+				var clip = Clip;
+				if (clip == null)
 				{
-					if (Clip != null)
-					{
-						return clippedFrame.IntersectWith(Clip.Rect);
-					}
-					return clippedFrame;
+					return !needsClipToSlot ? (Rect?) null : clippedFrame;
 				}
-
-				return Clip?.Rect;
+				return clip.Rect;
```

Comment:
This is no longer returning the intersection with `clippedFrame` when `Clip != null`, is that intentional?

Reply:
Yep it's part of the fix.